File: ios/StillMomentTests/Mocks/MockAttunementResolver.swift
//
//  MockAttunementResolver.swift
//  Still Moment
//
//  Test mock for AttunementResolverProtocol
//

import Foundation
@testable import StillMoment

final class MockAttunementResolver: AttunementResolverProtocol {
    // MARK: - Configurable returns

    var stubbedResolveResults: [String: ResolvedAttunement] = [:]
    var stubbedAudioURLs: [String: URL] = [:]
    var stubbedAllAvailable: [ResolvedAttunement] = []
    var shouldThrowOnResolveURL: Error?

    // MARK: - Recorded calls

    var resolvedIds: [String] = []
    var resolvedAudioURLIds: [String] = []

    // MARK: - Protocol

    func resolve(id: String) -> ResolvedAttunement? {
        self.resolvedIds.append(id)
        return self.stubbedResolveResults[id]
    }

    func resolveAudioURL(id: String) throws -> URL {
        self.resolvedAudioURLIds.append(id)
        if let error = self.shouldThrowOnResolveURL { throw error }
        guard let url = self.stubbedAudioURLs[id] else {
            throw AudioServiceError.soundFileNotFound
        }
        return url
    }

    func allAvailable() -> [ResolvedAttunement] {
        self.stubbedAllAvailable
    }
}

File: android/app/src/test/kotlin/com/stillmoment/testutil/MockAttunementResolver.kt
package com.stillmoment.testutil

import com.stillmoment.domain.models.ResolvedAttunement
import com.stillmoment.domain.services.AttunementResolverProtocol

/**
 * Test mock for AttunementResolverProtocol.
 *
 * Returns pre-configured results for resolve() and allAvailable().
 */
class MockAttunementResolver(
    private val resolveResult: Map<String, ResolvedAttunement> = emptyMap(),
    private val allAvailableResult: List<ResolvedAttunement> = emptyList()
) : AttunementResolverProtocol {
    override fun resolve(id: String): ResolvedAttunement? = resolveResult[id]
    override fun allAvailable(): List<ResolvedAttunement> = allAvailableResult
}
